azurerm/data_source_azuread_application.go

```diff
@@ -85,13 +87,18 @@ func dataSourceArmAzureADApplicationRead(d *schema.ResourceData, meta interface{
 
 		application = resp
 	} else {
-		resp, err := client.ListComplete(ctx, "")
+
+		// use the name to find the Azure AD application
+		name := d.Get("name").(string)
+		filter := "displayName eq '" + name + "'"
```

Comment:
this'd be better as `filter := fmt.Sprintf("displayName eq '%q'", name)

Reply:
Makes sense! Added as requested.